hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java

```diff
@@ -196,13 +187,8 @@ public void readData(ObjectDataInput in) throws IOException {
                 ignore(ignored);
             }
         }
-        int propSize = in.readInt();
-        properties = createHashMap(propSize);
-        for (int k = 0; k < propSize; k++) {
-            String key = in.readString();
-            String value = in.readString();
-            properties.put(key, value);
-        }
+
+        properties = SerializationUtil.readMap(in);
```

Comment:
this change should be reverted - `readObject` (used in `readMap`) does not read the same serialized form as the one produced by `writeString` on the `writeData` side

Reply:
a5710bbe37ed471893b68af41ed0e36447bcdfe5